extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php

```diff
@@ -408,11 +408,18 @@ public function getData( $page = 1 ) {
 		$this->preloadFirstRevId( $comments );
 		$pagination = $this->doPagination( $countComments, count( $comments ), $page );
 
+		$commentTitle = Title::makeTitle(
+			MWNamespace::getTalk( $this->mTitle->getNamespace() ),
+			$this->mTitle->getDBkey() . '/' . ARTICLECOMMENT_PREFIX
+		);
+
+		$canComment = $commentTitle->userCan( 'create' );
+
 		return [
 			'avatar' => AvatarService::renderAvatar( $wg->User->getName(), 50 ),
 			'userurl' => AvatarService::getUrl( $wg->User->getName() ),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)